packages/x-date-pickers/src/PickersLayout/usePickerLayout.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import { useSlotProps } from '@mui/base/utils';
+import { unstable_composeClasses as composeClasses } from '@mui/utils';
+import { PickersActionBar, PickersActionBarAction } from '../PickersActionBar';
+import { PickersLayoutProps, SubComponents } from './PickersLayout.types';
+import { getPickersLayoutUtilityClass } from './pickersLayoutClasses';
+import { DateOrTimeView } from '../internals/models';
+import { BaseToolbarProps } from '../internals/models/props/toolbar';
+
+function toolbarHasView<TValue, TView extends DateOrTimeView>(
+  toolbarProps: BaseToolbarProps<TValue, TView> | any,
+): toolbarProps is BaseToolbarProps<TValue, TView> {
+  return toolbarProps.view !== null;
+}
+
+// TODO: Remove this functions. It get introduced to mark `value` prop in PickersLayoutProps as not required.
+// The true type should be
+// - For pickers value: TDate | null
+// - For rangepickers value: [TDate | null, TDate | null]
+function toolbarHasValue<TValue, TView extends DateOrTimeView>(
+  toolbarProps: BaseToolbarProps<TValue | undefined, TView> | any,
+): toolbarProps is BaseToolbarProps<TValue, TView> {
+  return toolbarProps.value !== undefined;
+}
+
```

Comment:
Since the `value: TValue` of layout props is interpreted as `Proptypes.any.require`, it returns an error when we pass `null`, which is a valid value for pickers (not for range pickers)

Since I did not find any solution to this problem, I did the following trick: override it with

```
value?: TValue
```

To be compliant with props types expected by the toolbar, I added this `toolbarHasValue` to ensure value is not undefined and set the correct type

@LukasTy @flaviendelangle  Do you agree with this nasty TS trick?

Reply:
I've got a question from different PoV: do we need `value: TValue` in toolbar instead of `value?: TValue`?
Pickers are fine working without any initial value defined. Do we need to force our selves so that internal components have to have this field defined marked as required if it causes us issues?
What do you think about the following diff?
```diff
diff --git a/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerToolbar.tsx b/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerToolbar.tsx
index e12140b14..5fe81985e 100644
--- a/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerToolbar.tsx
+++ b/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerToolbar.tsx
@@ -70,7 +70,7 @@ export const DateRangePickerToolbar = React.forwardRef(function DateRangePickerT
   const props = useThemeProps({ props: inProps, name: 'MuiDateRangePickerToolbar' });
 
   const {
-    value: [start, end],
+    value,
     isMobileKeyboardViewOpen,
     toggleMobileKeyboardView,
     rangePosition,
@@ -78,6 +78,7 @@ export const DateRangePickerToolbar = React.forwardRef(function DateRangePickerT
     toolbarFormat,
     className,
   } = props;
+  const [start, end] = value ?? [null, null];
 
   const localeText = useLocaleText<TDate>();
 
diff --git a/packages/x-date-pickers/src/PickersLayout/usePickerLayout.tsx b/packages/x-date-pickers/src/PickersLayout/usePickerLayout.tsx
index bd0a75b6a..5008ddd57 100644
--- a/packages/x-date-pickers/src/PickersLayout/usePickerLayout.tsx
+++ b/packages/x-date-pickers/src/PickersLayout/usePickerLayout.tsx
@@ -13,16 +13,6 @@ function toolbarHasView<TValue, TView extends DateOrTimeView>(
   return toolbarProps.view !== null;
 }
 
-// TODO: Remove this functions. It get introduced to mark `value` prop in PickersLayoutProps as not required.
-// The true type should be
-// - For pickers value: TDate | null
-// - For rangepickers value: [TDate | null, TDate | null]
-function toolbarHasValue<TValue, TView extends DateOrTimeView>(
-  toolbarProps: BaseToolbarProps<TValue | undefined, TView> | any,
-): toolbarProps is BaseToolbarProps<TValue, TView> {
-  return toolbarProps.value !== undefined;
-}
-
 const useUtilityClasses = (ownerState: PickersLayoutProps<any, any>) => {
   const { classes, isLandscape } = ownerState;
   const slots = {
@@ -104,10 +94,7 @@ const usePickerLayout = <TValue, TView extends DateOrTimeView>(
     ownerState: { ...props, wrapperVariant },
   });
   const toolbar =
-    toolbarHasView(toolbarProps) &&
-    toolbarHasValue<TValue, TView>(toolbarProps) &&
-    shouldRenderToolbar &&
-    !!Toolbar ? (
+    toolbarHasView(toolbarProps) && shouldRenderToolbar && !!Toolbar ? (
       <Toolbar {...toolbarProps} />
     ) : null;
 
diff --git a/packages/x-date-pickers/src/TimePicker/TimePickerToolbar.tsx b/packages/x-date-pickers/src/TimePicker/TimePickerToolbar.tsx
index 9cdda9ae5..b0f829dfb 100644
--- a/packages/x-date-pickers/src/TimePicker/TimePickerToolbar.tsx
+++ b/packages/x-date-pickers/src/TimePicker/TimePickerToolbar.tsx
@@ -125,7 +125,7 @@ export function TimePickerToolbar<TDate extends unknown>(inProps: TimePickerTool
   const {
     ampm,
     ampmInClock,
-    value,
+    value = null,
     isLandscape,
     isMobileKeyboardViewOpen,
     onChange,
diff --git a/packages/x-date-pickers/src/internals/models/props/toolbar.ts b/packages/x-date-pickers/src/internals/models/props/toolbar.ts
index 5acd32d80..f721169ab 100644
--- a/packages/x-date-pickers/src/internals/models/props/toolbar.ts
+++ b/packages/x-date-pickers/src/internals/models/props/toolbar.ts
@@ -5,7 +5,7 @@ export interface BaseToolbarProps<TValue, TView extends DateOrTimeView>
   extends ExportedBaseToolbarProps {
   isLandscape: boolean;
   onChange: (newValue: TValue) => void;
-  value: TValue;
+  value?: TValue;
   /**
    * Currently visible picker view.
    */

```
Or maybe even making the value not required in the original type?
```diff
diff --git a/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerToolbar.tsx b/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerToolbar.tsx
index e12140b14..5fe81985e 100644
--- a/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerToolbar.tsx
+++ b/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerToolbar.tsx
@@ -70,7 +70,7 @@ export const DateRangePickerToolbar = React.forwardRef(function DateRangePickerT
   const props = useThemeProps({ props: inProps, name: 'MuiDateRangePickerToolbar' });
 
   const {
-    value: [start, end],
+    value,
     isMobileKeyboardViewOpen,
     toggleMobileKeyboardView,
     rangePosition,
@@ -78,6 +78,7 @@ export const DateRangePickerToolbar = React.forwardRef(function DateRangePickerT
     toolbarFormat,
     className,
   } = props;
+  const [start, end] = value ?? [null, null];
 
   const localeText = useLocaleText<TDate>();
 
diff --git a/packages/x-date-pickers/src/PickersLayout/PickersLayout.types.ts b/packages/x-date-pickers/src/PickersLayout/PickersLayout.types.ts
index 21dc3ad5a..67b2f7e5d 100644
--- a/packages/x-date-pickers/src/PickersLayout/PickersLayout.types.ts
+++ b/packages/x-date-pickers/src/PickersLayout/PickersLayout.types.ts
@@ -70,8 +70,7 @@ export interface PickersLayoutSlotsComponentsProps<TValue, TView extends DateOrT
 }
 
 export interface PickersLayoutProps<TValue, TView extends DateOrTimeView>
-  extends Omit<UsePickerLayoutPropsResponseLayoutProps<TValue, TView>, 'value'> {
-  value?: TValue;
+  extends UsePickerLayoutPropsResponseLayoutProps<TValue, TView> {
   className?: string;
   children?: React.ReactNode;
   sx?: SxProps<Theme>;
diff --git a/packages/x-date-pickers/src/PickersLayout/usePickerLayout.tsx b/packages/x-date-pickers/src/PickersLayout/usePickerLayout.tsx
index bd0a75b6a..5008ddd57 100644
--- a/packages/x-date-pickers/src/PickersLayout/usePickerLayout.tsx
+++ b/packages/x-date-pickers/src/PickersLayout/usePickerLayout.tsx
@@ -13,16 +13,6 @@ function toolbarHasView<TValue, TView extends DateOrTimeView>(
   return toolbarProps.view !== null;
 }
 
-// TODO: Remove this functions. It get introduced to mark `value` prop in PickersLayoutProps as not required.
-// The true type should be
-// - For pickers value: TDate | null
-// - For rangepickers value: [TDate | null, TDate | null]
-function toolbarHasValue<TValue, TView extends DateOrTimeView>(
-  toolbarProps: BaseToolbarProps<TValue | undefined, TView> | any,
-): toolbarProps is BaseToolbarProps<TValue, TView> {
-  return toolbarProps.value !== undefined;
-}
-
 const useUtilityClasses = (ownerState: PickersLayoutProps<any, any>) => {
   const { classes, isLandscape } = ownerState;
   const slots = {
@@ -104,10 +94,7 @@ const usePickerLayout = <TValue, TView extends DateOrTimeView>(
     ownerState: { ...props, wrapperVariant },
   });
   const toolbar =
-    toolbarHasView(toolbarProps) &&
-    toolbarHasValue<TValue, TView>(toolbarProps) &&
-    shouldRenderToolbar &&
-    !!Toolbar ? (
+    toolbarHasView(toolbarProps) && shouldRenderToolbar && !!Toolbar ? (
       <Toolbar {...toolbarProps} />
     ) : null;
 
diff --git a/packages/x-date-pickers/src/TimePicker/TimePickerToolbar.tsx b/packages/x-date-pickers/src/TimePicker/TimePickerToolbar.tsx
index 9cdda9ae5..b0f829dfb 100644
--- a/packages/x-date-pickers/src/TimePicker/TimePickerToolbar.tsx
+++ b/packages/x-date-pickers/src/TimePicker/TimePickerToolbar.tsx
@@ -125,7 +125,7 @@ export function TimePickerToolbar<TDate extends unknown>(inProps: TimePickerTool
   const {
     ampm,
     ampmInClock,
-    value,
+    value = null,
     isLandscape,
     isMobileKeyboardViewOpen,
     onChange,
diff --git a/packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts b/packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts
index 077cabcea..564eb8496 100644
--- a/packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts
+++ b/packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts
@@ -203,7 +203,7 @@ export interface UsePickerValueViewsResponse<TValue> {
  * Props passed to `usePickerLayoutProps`.
  */
 export interface UsePickerValueLayoutResponse<TValue> extends UsePickerValueActions {
-  value: TValue;
+  value?: TValue;
   onChange: (newValue: TValue) => void;
 }
 
diff --git a/packages/x-date-pickers/src/internals/models/props/toolbar.ts b/packages/x-date-pickers/src/internals/models/props/toolbar.ts
index 5acd32d80..f721169ab 100644
--- a/packages/x-date-pickers/src/internals/models/props/toolbar.ts
+++ b/packages/x-date-pickers/src/internals/models/props/toolbar.ts
@@ -5,7 +5,7 @@ export interface BaseToolbarProps<TValue, TView extends DateOrTimeView>
   extends ExportedBaseToolbarProps {
   isLandscape: boolean;
   onChange: (newValue: TValue) => void;
-  value: TValue;
+  value?: TValue;
   /**
    * Currently visible picker view.
    */

```

Reply:
Thanks for the different PoV, I may have spent too much time on Layout typing that I forget I could modify others 👍

Since toolbar value is not controlled, it could be `undefined`, and that's already a better solution than what I proposed

But the cleanest solution could be about the props type generator. I don't get why it set `Required` to a type `any`. If it is of type `any`, it can contain `null` or `undefined` and so it's not required

Reply:
> But the cleanest solution could be about the props type generator. I don't get why it set `Required` to a type `any`. If it is of type `any`, it can contain `null` or `undefined` and so it's not required

You could look into [this solution](https://github.com/mui/mui-x/pull/7225#pullrequestreview-1226827528) suggested by Olivier. 🤔

Reply:
I like Olivier's solutio here